# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build fontconfig-build glib-build libpng-build pixman-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.67; am-1.11; lt-2.2.6b)
libtoolize -c -f; aclocal -I build; autoconf; autoheader -f; automake -a -c
./configure --disable-interpreter --disable-static --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install util/cairo-trace/cairo-trace $DESTDIR/fb/bin
install -d $DESTDIR/fb/include/cairo
install -m 644 cairo-version.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo-deprecated.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo-features.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo-ft.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo-pdf.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo-ps.h $DESTDIR/fb/include/cairo
install -m 644 src/cairo-svg.h $DESTDIR/fb/include/cairo
install -m 644 util/cairo-gobject/cairo-gobject.h $DESTDIR/fb/include/cairo
install -d $DESTDIR/fb/lib
install src/.libs/libcairo.so.2.* $DESTDIR/fb/lib/libcairo.so.2
ln -sf libcairo.so.2 $DESTDIR/fb/lib/libcairo.so
install util/cairo-gobject/.libs/libcairo-gobject.so.2.* $DESTDIR/fb/lib/libcairo-gobject.so.2
ln -sf libcairo-gobject.so.2 $DESTDIR/fb/lib/libcairo-gobject.so
install -d $DESTDIR/fb/lib/cairo
install util/cairo-trace/.libs/libcairo-trace.so.0.0.0 $DESTDIR/fb/lib/cairo/libcairo-trace.so
install -d $DESTDIR/fb/lib/pkgconfig
install -m 644 src/cairo.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 src/cairo-ft.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 src/cairo-gobject.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 src/cairo-png.pc $DESTDIR/fb/lib/pkgconfig

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/cairo-build.pc << EOF
Name: cairo
Version: 1.10.2
Description: Cairo drawing library
Requires: $REQUIRES

devel=\\
/fb/include/cairo.h \\
/fb/include/cairo-deprecated.h \\
/fb/include/cairo-features.h \\
/fb/include/cairo-ft.h \\
/fb/include/cairo-gobject.h \\
/fb/include/cairo-pdf.h \\
/fb/include/cairo-ps.h \\
/fb/include/cairo-svg.h \\
/fb/include/cairo-version.h \\
/fb/lib/libcairo.so \\
/fb/lib/libcairo-gobject.so \\
/fb/lib/pkgconfig/cairo.pc \\
/fb/lib/pkgconfig/cairo-ft.pc \\
/fb/lib/pkgconfig/cairo-gobject.pc \\
/fb/lib/pkgconfig/cairo-png.pc

exec=\\
/fb/bin/cairo-trace \\
/fb/lib/libcairo.so.2  \\
/fb/lib/libcairo-gobject.so.2 \\
/fb/lib/cairo/libcairo-trace.so
EOF

exit
--- cairo-1.10.2.orig/src/cairo-image-surface.c
+++ cairo-1.10.2/src/cairo-image-surface.c
@@ -297,7 +297,7 @@
 	ret = PIXMAN_a8;
 	break;
     case CAIRO_FORMAT_RGB24:
-	ret = PIXMAN_x8r8g8b8;
+	ret = PIXMAN_r8g8b8;
 	break;
     case CAIRO_FORMAT_RGB16_565:
 	ret = PIXMAN_r5g6b5;
@@ -684,7 +684,7 @@
     case CAIRO_FORMAT_ARGB32:
 	return 32;
     case CAIRO_FORMAT_RGB24:
-	return 32;
+	return 24;
     case CAIRO_FORMAT_RGB16_565:
 	return 16;
     case CAIRO_FORMAT_A8:
--- cairo-1.10.2.orig/src/cairo-png.c
+++ cairo-1.10.2/src/cairo-png.c
@@ -628,7 +628,7 @@
 	    break;
 
 	case PNG_COLOR_TYPE_RGB:
-	    format = CAIRO_FORMAT_RGB24;
+	    format = CAIRO_FORMAT_ARGB32;
 	    png_set_read_user_transform_fn (png, convert_bytes_to_data);
 	    break;
     }
